draft-ietf-moq-transport.md

```diff
@@ -742,76 +746,74 @@ done in the context of an established MoQT session.
-it MUST send an ANNOUNCE to any subscriber that has subscribed to ANNOUNCE for
-that namespace, or a more generic set including that namespace. A publisher MAY
-send the ANNOUNCE to any other subscriber.
-
-An endpoint SHOULD NOT, however, send an ANNOUNCE advertising a namespace that
-exactly matches a namespace for which the peer sent an earlier ANNOUNCE
-(i.e. an ANNOUNCE ought not to be echoed back to its sender).
-
-The receiver of an ANNOUNCE_OK or ANNOUNCE_ERROR SHOULD report this to the
-application to inform the search for additional subscribers for a namespace,
-or abandoning the attempt to publish under this namespace. This might be
-especially useful in upload or chat applications. A subscriber MUST send exactly
-one ANNOUNCE_OK or ANNOUNCE_ERROR in response to an ANNOUNCE. The publisher
-SHOULD close the session with a protocol error if it receives more than one.
-
-An UNANNOUNCE message withdraws a previous ANNOUNCE, although it is not a
-protocol error for the subscriber to send a SUBSCRIBE or FETCH message for a
-track in a namespace after receiving an UNANNOUNCE.
-
-A subscriber can send ANNOUNCE_CANCEL to revoke acceptance of an ANNOUNCE, for
-example due to expiration of authorization credentials. The message enables the
-publisher to ANNOUNCE again with refreshed authorization, or discard associated
-state. After receiving an ANNOUNCE_CANCEL, the publisher does not send
-UNANNOUNCE.
-
-While ANNOUNCE indicates to relays how to connect publishers and subscribers, it
+## Sending Publish Messages
+
+A non-relay publisher MAY send a PUBLISH message to any subscriber. PUBLISH
+indicates to the subscriber that the publisher has a single track available, or
+one or more tracks available in a namespace. A subscriber MAY send SUBSCRIBE or
+FETCH for a track without having received an PUBLISH for the tracks namespace.
+
+If a publisher is authoritative for a given track or namespace, or is a relay
+that has received an authorized PUBLISH for that namespace from an upstream
+publisher, it MUST send a PUBLISH to any subscriber that has subscribed to that
+namespace, or a more generic set including that namespace. A publisher MAY send
+the PUBLISH to any other subscriber, but a relay MUST NOT send a PUBLISH without
+an authorized SUBSCRIBE first.
+
+An endpoint SHOULD NOT, however, send a PUBLISH advertising a namespace or track
+that exactly matches a namespace for which the peer sent an earlier PUBLISH
+(i.e. a PUBLISH ought not to be echoed back to its sender).
+
+The receiver of an PUBLISH_OK or PUBLISH_ERROR SHOULD report this to the
+application. A subscriber MUST send exactly one PUBLISH_OK or PUBLISH_ERROR in
+response to a PUBLISH. The publisher SHOULD close the session with a protocol
+error if it receives more than one.
+
+A SUBSCRIBE_DONE message terminates a PUBLISH of a track or of a namespace.
+
+A subscriber can send UNSUBSCRIBE to revoke acceptance of a PUBLISH, for example
```

Comment:
After the first clause, I'm finding this paragraph confusing.  Why would the subscriber unsubscribe because its credentials expired?  Wouldn't they issue a SUBSCRIBE_UPDATE to update their credentials?

Also, I thought we did send a SUBSCRIBE_DONE in response to UNSUBSCRIBE for SUBSCRIBE?

Reply:
> Why would the subscriber unsubscribe because its credentials expired?

In this case it's the publisher's credentials that expired.

> Also, I thought we did send a SUBSCRIBE_DONE in response to UNSUBSCRIBE for SUBSCRIBE?

No, SUBSCRIBE_DONE is not sent in response to UNSUBSCRIBE anymore.

Reply:
Reminds me that there are some issues open to write down the SUBSCRIBE state machine.  Maybe it's stable enough now?